Osd

rsurpur20 opened the pull request.

victorhuangwq commented on src/controllers/osd.ts:
```diff
@@ -0,0 +1,57 @@
+import nconf from 'nconf';
+import { NextFunction } from 'express';
+import plugins from '../plugins';
+import metaConfig from '../meta/configs';
```
looking at the original file, you didn't seemed to have imported meta properly

victorhuangwq commented on src/controllers/osd.ts:
```diff
@@ -0,0 +1,57 @@
+
+export default function handler(req, res, next:NextFunction): void {
+    if (plugins.hooks.hasListeners('filter:search.query')) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
```
this comment should be above the plugins, not on res.type

rsurpur20 replied:
I got a linter error when I try to move it anywhere else

rsurpur20 commented:
Below is an outline of the changes I made, bugs I ran into, and information about my submission. 

Changes: There are two main changes: translating the osd.js file to typescript and updating the test suite. For the translation, I referred to osd.js, studied the logic, studied typescript syntax, and made the appropriate translation. I also installed a new XML type from this site: https://www.npmjs.com/package/@types/xml. For the test suite, I added done() to the end of some functions that were timing out or throwing assertion errors.

Common Problems:
Types: I had an issue configuring the return type of generate XML (). The appropriate return type was a string, but even after I specified the return type as string, it was read as type any. With the help of the TA’s, we changed the syntax to save the xml() method call as a string and then return that string. 
I got an error, “”before all” hook for “should correctly compare a password and a hash”: Error: Route.get() requires a callback function but got a [object Undefined]”. Working with Victor and Tak Ho for several hours, we removed old versions of node, reinstalled the new version of node, updated the bash profile, and removed the “default” in “export default function” while also disabling the linter on that line to appropriately remove the “default” keyword. 

Submission Information:

Lint: My linter passes locally, but not on GitHub meaning there is a difference in the versions of modules or dependencies. While GitHub says it has failed, it would require a deep look into the config file to find the difference between GitHub’s versions and mine. 

Tests: Right now I am passing 794 tests with 55.86% statement coverage, 37.11% branch coverage, 50.44% function coverage, and 56.27% line coverage. If given more time and resources, my goal would be to increase these numbers. It currently fails on timeout issues and assertion errors external to my contribution to changing the osd.js file.
